.gitlab-ci.yml

```diff
@@ -0,0 +1,215 @@
+# Copyright 2018 Red Hat, Inc.
+# Author: Jan Pokorny <jpokorny@redhat.com>
+# Part of pacemaker project
+# SPDX-License-Identifier: FSFAP
+
+variables:
+  GIT_DEPTH: "16"
+
+stages:
+  - buildenv
+  - build
+  - test
+
+.fedora-rawhide: &fedora-rawhide-anchor
+  tags:
+    - docker
+  image: fedora:rawhide
+
+.buildenv: &buildenv-anchor
+  stage: buildenv
+  variables:
+    GIT_STRATEGY: none
+
+.compilation: &compilation-anchor
+  only:
+   - pushes
+  # only since GitLab 11.4
+  #  changes:
+  #    - "configure.ac"
+  #    - "**/*.[ch]"
+  #    - "**/*.xml"
+  #    - "**/*.xsl"
+  #    - "**/Makefile.am"
+  #    - ".gitlab-ci.yml"
+
+buildenv:fedora-rawhide:info:
+  # primarily to gather some info about the image itself/container settings
+  <<: *buildenv-anchor  #extends: .buildenv
+  <<: *fedora-rawhide-anchor  #extends: .fedora-rawhide
+  script:
+   - cat /proc/meminfo | grep Free
+   - df -H
+   - lsblk
+   - nproc
+   - ulimit -a
+   - dnf --version
+
+buildenv:fedora-rawhide:rpms:
+  # primarily to download RPMs, but alas, "download" plugin not baked into image
+  <<: *buildenv-anchor  #extends: .buildenv
+  <<: *fedora-rawhide-anchor  #extends: .fedora-rawhide
+  cache:
+    key: "fedora-rawhide-dnf"
+    paths:
+      - ".cache/fedora-rawhide/dnf/rawhide-*"
+  script:
+    - mkdir -p .cache/fedora-rawhide/dnf
+    - cd .cache/fedora-rawhide
+    - mv dnf/rawhide-* /var/cache/dnf 2>/dev/null || true
+    - dnf-custom() { timeout 192 dnf -y --noplugins --nodocs
+                                 --setopt=tsflags=test
+                                 --setopt=install_weak_deps=0
+                                 --setopt=keepcache=1
+                                 --setopt=autocheck_running_kernel=0
+                                 --disablerepo='*' --enablerepo=rawhide "$@"; };
+      dnf-custom install
+        autoconf automake pkgconf-pkg-config libtool libtool-ltdl-devel
+        make findutils
+        gcc clang compiler-rt
+        libuuid-devel libxslt-devel bzip2-devel
+        'pkgconfig(glib-2.0) >= 2.16.0' 'pkgconfig(libxml-2.0)'
+        'pkgconfig(libqb) >= 0.13' 'pkgconfig(libcpg)'
```

Comment:
Not directly relevant to this PR, just mentioning: I've been thinking it might be worthwhile to define configure variables for minimum versions of dependencies, so we can ensure we keep them consistent between configure.ac, pacemaker.spec.in, and anywhere else like this that might need them. Slightly complicated by pacemaker.spec having a .in that's not currently processed by configure, but GNUmakefile targets.

Reply:
Or a simple, consistent file of shell-parsable assignments, ad-hoc
`grep`able, `eval`able etc. that would be single point of authority?
Could be source in `configure` script, itself placed in a side dir
(`maint`?) and referred to from `INSTALL.md` that could hence
be more static.

Because it's harder to extract from `configure.ac` than other way
around.

Reply:
I was thinking of substitution variables aka output variables, so that anything that needed them would be a .in file and use %VAR%. The values themselves could be directly in configure.ac or in an included .m4 file.

For the spec, I'm thinking it would benefit from an additional level of indirection, though a bit more confusing. pacemaker.spec.template.in would be the source, configure would create pacemaker.spec.template, and GNUmakefile would use that to create pacemaker-DISTRO.spec.

Reply:
Well, I suggest getting the feeling of `meson` + `ninja` (e.g. systemd),
and you won't ever want to look back at autotools, for a noticable
difference in both simplicity and speed.

In that light, I would be worry to overload `configure.ac` as even
more central point, it shall preferably be on the contrary when
a generic reusability is possible, and what I proposed is along
those lines.